pages/stats/challenges.tsx

```diff
@@ -0,0 +1,188 @@
+              <Grid.Col key={row.id} span={1}>
+                <Accordion p={0} chevronPosition="right" variant="separated">
+                  <Accordion.Item value={row.id} key={row.name}>
+                    <Accordion.Control>
+                      <AccordionLabel {...row} />
+                    </Accordion.Control>
+                    <Accordion.Panel>
+                      <AccordionContent sbpsData={sbpsData} challenge={row} />
+                    </Accordion.Panel>
+                  </Accordion.Item>
+                </Accordion>
+              </Grid.Col>
+            );
+          })}
+        </Grid>
+      </Container>
+    </>
+  );
+};
+
+function AccordionLabel({ name, description, reward }: ChallengesType) {
+  return (
+    <Card p="lg" radius="md" withBorder>
+      <Flex direction="column" gap={4}>
+        <Title order={3} color="orange.5">
+          {name}
+        </Title>
+        <Text size="md">{description}</Text>
+        <Group spacing={"xs"}>
+          <IconMedal size={16} />
+          <Text size="sm" color="blue.4">
+            {reward} Merits
+          </Text>
+        </Group>
+      </Flex>
+    </Card>
+  );
+}
+
+interface AccordionContentProps {
+  sbpsData: SbpsType[];
+  challenge: ChallengesType;
+}
+
+function AccordionContent({ challenge, sbpsData }: AccordionContentProps) {
+  const counters = challenge.counters;
+  const mappedSources = counters.sources.map((id) => ({
+    id,
+    squad: sbpsData.find((x) => x.id === id),
+  }));
+  const mappedTargets = counters.targets.map((id) => ({
+    id,
+    squad: sbpsData.find((x) => x.id === id),
+  }));
+  return (
+    <Stack>
+      <Title order={5} color="red.5">
+        Counters
+      </Title>
+      {counters.research.length ? (
+        <List spacing="sm">
+          {...counters.research.map((id) => {
+            return <List.Item key={id}>{id}</List.Item>;
+          })}
+        </List>
+      ) : (
+        <></>
+      )}
+      {counters.spawnee.length ? (
+        <List spacing="sm">
+          {...counters.spawnee.map((id) => {
+            return <List.Item key={id}>{id}</List.Item>;
+          })}
+        </List>
+      ) : (
+        <></>
+      )}
+      {mappedSources.length ? (
+        <List spacing="sm">
+          {...mappedSources.map((item) => (
+            <List.Item key={item.id}>{item.squad?.ui.screenName || "---"}</List.Item>
+          ))}
+        </List>
+      ) : (
+        <></>
+      )}
+      {mappedTargets.length ? (
+        <List spacing="sm">
+          {...mappedTargets.map((item) => (
+            <List.Item key={item.id}>{item.squad?.ui.screenName || "---"}</List.Item>
+          ))}
+        </List>
+      ) : (
+        <></>
+      )}
+    </Stack>
+  );
+}
+
+export const getStaticProps = async () => {
+  const { dailyChallengesData, weeklyChallengesData, sbpsData } = await getMappings();
+
+  return {
+    props: {
+      dailyChallengesData,
+      weeklyChallengesData,
+      sbpsData,
```

Comment:
Btw it would be great if we wouldn't pass the full data here to the client side of the component. Because it works great when you visit the site the first time. But when you just click on the site, it passes this data to the client, where it's rendered. 

And when you pass the data like this, it's a lot of data, for example for our races pages, it's 5MB json file. You can see it during the build. 

<img width="866" alt="image" src="https://github.com/cohstats/coh3-stats/assets/8086995/5a7b568c-e24a-4c55-83bd-daa15d502db7">

So ideally here in the function `getStaticProps` you precompute all the data which are necessary for your components and pass it there.

Reply:
Ummm not sure how to tackle that part 😢